R/download_springer_book.R

```diff
@@ -0,0 +1,167 @@
+
+      springer_table <-
+        readxl::read_excel(tf) %>%
+        janitor::clean_names()
+
+    }
+
+    book_info <-
+      springer_table %>%
+      dplyr::filter(book_title == book) %>%
+      dplyr::filter(copyright_year == max(copyright_year))
+
+    edition <-
+      book_info %>%
+      dplyr::pull(edition) %>%
+      unique()
+
+    en_book_type <-
+      book_info %>%
+      dplyr::pull(english_package_name) %>%
+      unique()
+
+    clean_book_title <-
+      gsub('/', '-', book)
+
+    message(paste0('Downloading ', clean_book_title))
+
+    dir.create(
+      path = sprintf('%s/%s',
+                     destination_folder,
+                     en_book_type),
+      showWarnings = FALSE,
+      recursive = TRUE
+    )
+
+    if(nrow(book_info) == 1) {
+
+      if(file.exists(
+        sprintf(
+          '%s/%s/%s - %s.pdf',
+          destination_folder,
+          en_book_type,
+          clean_book_title,
+          edition)
+      )) {
+
+        message('File already exists, skipping book download')
+
+        return()
+
+      } else {
+
+        download_url <-
+          book_info %>%
+          dplyr::pull(open_url) %>%
+          httr::GET() %>%
+          magrittr::extract2('url') %>%
+          stringr::str_replace('book', paste0('content', '/', 'pdf')) %>%
+          stringr::str_replace('%2F', '/') %>%
+          paste0('.pdf')
+
+        download.file(
+          url = download_url,
+          destfile = sprintf(
+            '%s/%s/%s - %s.pdf',
+            destination_folder,
+            en_book_type,
+            clean_book_title,
+            edition
+          ),
+          quiet = TRUE
+        )
+
+        Sys.sleep(1)
+
+      }
+
+    } else {
+
+      list_of_downloads <-
+        book_info %>%
+        dplyr::pull(open_url)
+
+      purrr::map(
+        list_of_downloads,
+        function(book_url) {
+
+          url_info <-
+            book_url %>%
+            httr::GET()
+
+          download_url <-
+            url_info %>%
+            magrittr::extract2('url') %>%
+            stringr::str_replace('book', paste0('content', '/', 'pdf')) %>%
+            stringr::str_replace('%2F', '/') %>%
+            paste0('.pdf')
+
+          subtitle <-
+            url_info %>%
+            httr::content() %>%
+            rvest::html_nodes(xpath = '//*[@id="main-content"]/article[1]/div/div/div[1]/div/div/div[1]/div[2]/h2') %>%
+            rvest::html_text()
+
+          download.file(
+            url = download_url,
+            destfile = sprintf(
+              '%s/%s/%s - %s - %s.pdf',
+              destination_folder,
+              en_book_type,
+              clean_book_title,
+              subtitle,
+              edition
+            ),
+            quiet = TRUE
+          )
+
+          Sys.sleep(1)
+
+        }
+
+      )
+
+    }
+
+    t2 <- Sys.time()
+
+    return_time <- t2 - t1
+
+    message(paste0('Time Elapsed: ', format(return_time, digits = 2)))
```

Comment:
Could you please also add the book title in the message here?

Reply:
Sure, no problems. I'll be adding that in the next commits